haystack/schema.py

```diff
@@ -327,6 +327,42 @@ def from_json(cls, data):
         return cls.from_dict(data)
 
 
+@dataclass
+class AudioAnswer(Answer):
+    answer: Path
+    context: Optional[Path] = None
+    offsets_in_document: Optional[Any] = None
+    offsets_in_context: Optional[Any] = None
+
+    def __str__(self):
+        return f"<AudioAnswer: answer='{self.answer}', score={self.score}, context='{self.context}'>"
+
+    def __repr__(self):
+        return f"<AudioAnswer {asdict(self)}>"
+
+
+@dataclass
+class GeneratedAudioAnswer(AudioAnswer):
+    type: str = "text-to-speech"
```

Comment:
Good catch! I will review how the type is used. I originally used `generative` to mimick normal `Answers`, but then I realized this is an even more specific type, so I went for `text-to-speech`.

For now the content of this field is just a label for users to know if the content of the object is extracted or generated, but I don't know how we want to manage them in the future. I can stick to `generative` for now.

Reply:
What other types could we imagine in future use cases? `extractive` as the `type: Literal["generative", "extractive", "other"]` in `Answer`? Or do you have anything else in mind? If we can come up with some possible values, I would keep the field. Otherwise we might better remove it and then add it again in case a use case comes up in future.